### ISSUE TITLE
SES email client doesn't use "from" name field

### ISSUE THREAD

**Issue by GeitV** [micronaut-projects/micronaut-email#117]
### Expected Behavior

When sending email through SES and setting `micronaut.email.from.name` in application config to "Test LTD" (or through `Email` field **to**), then we should see in our email inbox that the email was coming from "**Test LTD**".
(`"from": "Test LTD <noreply@test.com>"`)

### Actual Behaviour

Email comes from **noreply**. Seems like **name** field is ignored, similar to [JavaMail issue](https://github.com/micronaut-projects/micronaut-email/pull/59).

**NOTE**: this does work correctly, when the Email has attachments added. These have the right name as sender. I don't know how it differs internally, but it does.

### Steps To Reproduce

Create project with `application.yaml` config as:
```yaml
micronaut:
  email:
    from:
      email: noreply@test.com
      name: Test LTD
```
or, when using `Contact` on the `Email.from` field
```kotlin
val email = Email.builder()
    .to(to)
    .from(Contact("noreply@test.com", "Test LTD"))
```

Add gradle dependency
```kotlin
    // mail service
    implementation("io.micronaut.email:micronaut-email-amazon-ses") {
        exclude("software.amazon.awssdk", "apache-client")
        exclude("software.amazon.awssdk", "netty-nio-client")
    }

    implementation("io.micronaut.aws:micronaut-aws-sdk-v2")
    implementation("software.amazon.awssdk:url-connection-client")
```

Send the email without any attachments and tada 🎉 

### Environment Information

- kotlin 17
- micronaut-email-amazon-ses 1.3.1
- not native

### Version

3.6.1

**Comment by KristjanErik** [comment]
Any progress on this?

**Comment by sdelamo** [comment]
We are setting the `from` here: 

https://github.com/micronaut-projects/micronaut-email/blob/master/email-amazon-ses/src/main/java/io/micronaut/email/ses/SesEmailComposer.java#L98

Javadoc from `SendEmailRequest::source` says: 

> The email address that is sending the email. This email address must be either individually verified with Amazon SES, or from a domain that has been verified with Amazon SES. For information about verifying identities

It seems we are correctly passing to the method the `from` contact's email address. 

Not sure how we can provide the name as well.

**Comment by GeitV** [comment]
This StackOverflow thread seems to say that having the name in `source` field is valid:
https://stackoverflow.com/a/70434076

> try to change
> 
> ```Source: "noreply@domain_name.com",```
> 
> to
> 
>  ```Source: "Friendly Name <noreply@domain_name.com>",```

I'd suppose this would work with the Java SDK as well, but haven't tested it

**Comment by sdelamo** [comment]
> This StackOverflow thread seems to say that having the name in `source` field is valid: https://stackoverflow.com/a/70434076
> 
> > try to change
> > `Source: "noreply@domain_name.com",`
> > to
> > `Source: "Friendly Name <noreply@domain_name.com>",`
> 
> I'd suppose this would work with the Java SDK as well, but haven't tested it

But the AWS SES SDK says email address.

**Comment by GeitV** [comment]
As sending with attachment works, why not use the same method for sending email with or without attachment?
I'm looking at this part:
https://github.com/micronaut-projects/micronaut-email/blob/6b14741c9fcc05c653b98f2381f072315af2a439/email-amazon-ses/src/main/java/io/micronaut/email/ses/SesEmailComposer.java#L64-L73

**Comment by sdelamo** [comment]
Please note that you can workaround this. 

You can create a class such as: 

```java
@Singleton
@Replaces(SesEmailComposer.class) 
public class SesEmailComposerReplacement extends SesEmailComposer {
  public SesEmailComposer(MessageComposer messageComposer) {
        this.messageComposer = messageComposer;
    }

   @NonNull
    @Override
    public SesRequest compose(@NonNull @NotNull @Valid Email email) throws EmailException {
        if (CollectionUtils.isEmpty(email.getAttachments())) {
SendEmailRequest.Builder requestBuilder = SendEmailRequest.builder()
                .destination(destinationBuilder(email).build())
                .source() //DO CHANGES
                .message(message(email));
        if (email.getReplyTo() != null) {
            requestBuilder = requestBuilder.replyToAddresses(email.getReplyTo().getEmail());
        }
        return requestBuilder.build();
        }
        try {
            return sendRawEmailRequest(email);
        } catch (MessagingException | IOException e) {
            throw new EmailException(e);
        }
    }
}
``

**Comment by GeitV** [comment]
Ofcourse! Such a logical workaround. Did not think about replacing the bean, thanks @sdelamo - I hope this still gets resolved as well.

**Comment by sdelamo** [comment]
> Ofcourse! Such a logical workaround. Did not think about replacing the bean, thanks @sdelamo.

I am trying to get clarification from the AWS SES team about how this should be used.

**Comment by mvelez-nydig** [comment]
My team ran into this issue recently. We tried doing the bean replacement workaround, but the replacement bean never gets picked up for some reason (it still uses the original `SesEmailComposer`). 
I don't know what's the latest on this, but I have a possible fix, similar to what was done in https://github.com/micronaut-projects/micronaut-email/pull/59, PR coming soon.

**Comment by sdelamo** [comment]
I think this is solved already.

**Comment by MolyAgarwal** [comment]
@GeitV Is this issue solved for you? I am still facing the same issue.

**Comment by GeitV** [comment]
@MolyAgarwal we instead started using Postmark (for other features in it), so sadly I have no clue

**Comment by sdelamo** [comment]
> @GeitV Is this issue solved for you? I am still facing the same issue.
> CC: @sdelamo @KristjanErik

@MolyAgarwal Yes, it should be fixed. Which version of Micronaut Framework are you using?

**Comment by MolyAgarwal** [comment]
> > @GeitV Is this issue solved for you? I am still facing the same issue.
> > CC: @sdelamo @KristjanErik
> 
> @MolyAgarwal Yes, [it should be fixed](https://github.com/micronaut-projects/micronaut-email/pull/136/files). Which version of Micronaut Framework are you using?

I have upgraded micronaut version to 3.7.4 and it fixed for me also. Thanks @sdelamo .

**Comment by richmidwinter** [comment]
I'm on 4.1.2 and I think there might have been a regression here.

The fix in #136 seems to be absent from [master.](https://github.com/micronaut-projects/micronaut-email/blob/master/email-amazon-ses/src/main/java/io/micronaut/email/ses/SesEmailComposer.java#L97)

**Comment by sdelamo** [comment]
@richmidwinter thanks for reporting. I have merged up and I will release patch versions soon.
